tests/python_package_test/test_engine.py

```diff
@@ -739,3 +739,57 @@ def test_mape_dart(self):
         pred = gbm.predict(X)
         pred_mean = pred.mean()
         self.assertGreater(pred_mean, 18)
+
+    @staticmethod
+    def test_constant_features(y_true=None, expected_pred=None, more_params=None):
+        if y_true is not None and expected_pred is not None:
+            X_train = np.ones((len(y_true), 1))
+            y_train = np.array(y_true)
+            params = {
+                'objective': 'regression',
+                'num_class': 1,
+                'verbose': -1,
+                'min_data': 1,
+                'num_leaves': 2,
+                'learning_rate': 1,
+                'min_data_in_bin': 1,
+                'boost_from_average': True
+            }
+            params.update(more_params)
+            lgb_train = lgb.Dataset(X_train, y_train, params=params)
+            gbm = lgb.train(params, lgb_train,
+                            num_boost_round=2)
+            pred = gbm.predict(X_train)
+            for i in range(pred.shape[0]):
+                np.testing.assert_almost_equal(pred[i], expected_pred)
```

Comment:
It's possible to omit the loop and write just
```
 np.testing.assert_almost_equal(pred, expected_pred) 
```

Reply:
I tried this, but it cannot work for multi-class, which compares 2d with 1d.

Reply:
Oops, sorry. Then it's possible to write the following:
```
self.assertTrue(np.allclose(pred, expected_pred))
```